api/src/pcapi/core/offers/validation.py

```diff
@@ -100,15 +100,33 @@ def check_stock_quantity(quantity: int | None, bookingQuantity: int = 0) -> None
         raise errors
 
 
-def check_stock_price(price: decimal.Decimal, offer: models.Offer) -> None:
+def check_stocks_price(
+    list_stocks: list[serialization.StockCreationBodyModel] | list[serialization.StockEditionBodyModel],
+    offer: models.Offer,
+) -> None:
+    price_categories = {price_category.id: price_category for price_category in offer.priceCategories}
+    for stock in list_stocks:
+        check_stock_has_price_or_price_category(offer, stock, price_categories)
+        if stock.price_category_id:
+            error_msg = "priceCategoryId"
+            price = price_categories.get(stock.price_category_id).price  # type: ignore [union-attr]
```

Comment:
En utilisant `get()`, si l'élément n'est pas dans `price_categories`, alors `price_categories.get()` renverra `None`, qui n'a pas d'attribut `price`. On sait que ça ne peut pas arriver car on a appelé `check_stock_has_price_or_price_category()` avant, mais mypy ne le sait pas, lui, d'où son avertissement. Si on écrit l'accès avec des crochets, on ne devrait plus avoir l'avertissement de mypy.

```suggestion
            price = price_categories[stock.price_category_id].price  # type: ignore [union-attr]
```